batch/batch/worker/worker.py

```diff
@@ -1787,59 +1945,113 @@ def scoped_ensure_future(coro_or_future, *, loop=None) -> Iterator[asyncio.Futur
+            },
+            {
+                'source': '/hail-jars',
+                'destination': '/hail-jars',
+                'type': 'none',
+                'options': ['rbind', 'rw'],
+            },
+            {
+                'source': root_dir,
+                'destination': root_dir,
+                'type': 'none',
+                'options': ['rbind', 'rw'],
+            },
+            {
+                'source': '/batch',
+                'destination': '/batch',
+                'type': 'none',
+                'options': ['rbind', 'rw'],
+            },
+        ]
+
+        fs = LocalAsyncFS(pool)  # worker does not have a fs when initializing JVMs
+
+        c = Container(
+            fs=fs,
+            name=f'jvm-{index}',
+            image=Image(BATCH_WORKER_IMAGE, JVMUserCredentials(), client_session, pool),
+            scratch_dir=f'{root_dir}/container',
+            command=command,
+            cpu_in_mcpu=1000,
+            memory_in_bytes=worker_memory_per_core_bytes(CLOUD, instance_config.worker_type()),
+            env=['HAIL_WORKER_OFF_HEAP_MEMORY_PER_CORE_MB=2220'],
+            volume_mounts=volume_mounts,
         )
-        stop_event = asyncio.Event()
-        return cls(process, stop_event)
-
-    def __init__(self, process, stop_event: asyncio.Event):
-        self.process = process
-        self.stop_event = stop_event
-        self.buf = bytearray()
-        assert process.stdout is not None
-        self.stdout_pump = asyncio.ensure_future(self.pump_to_buffer(process.stdout))
-        assert process.stderr is not None
-        self.stderr_pump = asyncio.ensure_future(self.pump_to_buffer(process.stderr))
-
-    async def pump_to_buffer(self, strm: asyncio.StreamReader):
-        with scoped_ensure_future(self.stop_event.wait()) as stop_fut:
-            while not strm.at_eof() and not self.stop_event.is_set():
-                with scoped_ensure_future(strm.readline()) as read_fut:
-                    await asyncio.wait([read_fut, stop_fut], return_when=asyncio.FIRST_COMPLETED)
-                    if read_fut.done():
-                        result = read_fut.result()
-                        self.buf.extend(result)
-
-    def output(self) -> str:
-        return self.buf.decode()
-
-    def retrieve_and_clear_output(self) -> str:
-        buf = self.buf.decode()
-        self.buf = bytearray()
-        return buf
-
-    def kill(self):
-        return self.process.kill()
+
+        await c.create()
+        await c.start()
+
+        return JVMContainer(c, fs)
+
+    def __init__(self, container: Container, fs: LocalAsyncFS):
+        self.container = container
+        self.fs = fs
+        self.log_offset = 0
+
+    async def retrieve_and_clear_output(self) -> str:
+        logs = await self.get_log()
+        self.log_offset += len(logs)
+        return logs
+
+    async def get_log(self):
+        return await self.container.get_log(offset=self.log_offset)
```

Comment:
Because you successfully fixed the Hail Query logging problem already, I don't think we need offset at all. The JVMEntryway stdout & stderr are never displayed in the UI. Longer term, we should add a worker status page that shows the JVM logs, but that can show the *entire* log. I think we can completely rip out the offset stuff now.

Reply:
In fact we can rip out all of the JVM log stuff because there's no way to access it right now.

Reply:
It's used when logging error messages for the JVM. Can you double check you don't want this functionality?

Reply:
If you don't like the last commit, then I can revert it.